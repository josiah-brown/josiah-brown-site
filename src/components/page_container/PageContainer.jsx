import React from "react";

const PageContainer = (props) => {
  return (
    <div
      data-testid="page-container"
      className="relative adjusted_max_h bg-white dark:bg-neutral-800"
    >
      {props.children}
    </div>
  );
};

export default PageContainer;
