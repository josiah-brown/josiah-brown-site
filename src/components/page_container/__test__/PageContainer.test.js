import { cleanup, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import PageContainer from "../PageContainer";

afterEach(cleanup);

test("renders page container without crash", () => {
  render(<PageContainer></PageContainer>);
});

test("renders page container children without affecting them", () => {
  const { getByTestId } = render(
    <PageContainer>
      <div>Hello World</div>
    </PageContainer>
  );
  expect(getByTestId("page-container")).toContainHTML("<div>Hello World</div>");
});

test("matches page container snapshot 1", () => {
  const tree = renderer
    .create(
      <PageContainer>
        <div>
          <p>child 1</p>
          <p>child 2</p>
        </div>
      </PageContainer>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
