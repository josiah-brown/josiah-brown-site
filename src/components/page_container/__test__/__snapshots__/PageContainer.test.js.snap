// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches page container snapshot 1 1`] = `
<div
  className="relative adjusted_max_h bg-white dark:bg-neutral-800"
  data-testid="page-container"
>
  <div>
    <p>
      child 1
    </p>
    <p>
      child 2
    </p>
  </div>
</div>
`;
